File: src/Learn/Arrays101/FindMaxConsecutiveOnes.cs
public class Solution {
    public int FindMaxConsecutiveOnes(int[] nums) {
    	int maxResult = 0;
    	int currResult = 0;
        foreach(int num in nums){
        	if(num == 1){
        		currResult++;
        		maxResult = Math.Max(maxResult, currResult);
        	}
        	else{
        		maxResult = Math.Max(maxResult, currResult);
        		currResult = 0;
        	}
        	
        }
        return maxResult;
    }
}

File: src/Learn/Arrays101/findMaxConsecutiveOnes.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
    	int currCount = 0;
    	int maxCount = 0;
        for(auto num : nums){
        	if(num == 1) currCount++;
        	else{
        		maxCount = max(maxCount, currCount);
        		currCount = 0;
        	}
        }
        return max(maxCount, currCount);
    }
};